evm-ext/src/type_utils.rs

```diff
@@ -43,6 +89,39 @@ pub fn code_hash_struct_tag(code_hash: &B256) -> StructTag {
     }
 }
 
+pub fn get_move_account_nonce(
+    address: &AccountAddress,
+    resolver: &dyn MoveResolver,
+) -> Option<u64> {
+    let tag: &StructTag = &ACCOUNT_STRUCT_TAG;
+    let metadata = resolver.get_module_metadata(&tag.module_id());
+    let bytes = resolver
+        .get_resource_bytes_with_metadata_and_layout(address, tag, &metadata, Some(&ACCOUNT_LAYOUT))
```

Comment:
Would this fail when the account doesn't exist? Ethereum would default to zero `0`.

Reply:
This returns `None` when the account does not exist. On the EVM side it does still default to 0.